core/src/apps/management/recovery_device/layout.py

```diff
@@ -72,41 +69,60 @@
         else:
             word = await ctx.wait(keyboard)
 
-        if mnemonic_type == mnemonic.TYPE_SLIP39 and mnemonics:
-            if not advanced_shamir:
-                # check if first 3 words of mnemonic match
-                # we can check against the first one, others were checked already
-                if i < 3:
-                    share_list = mnemonics[0].split(" ")
-                    if share_list[i] != word:
-                        raise IdentifierMismatchError()
-                elif i == 3:
-                    for share in mnemonics:
-                        share_list = share.split(" ")
-                        # check if the fourth word is different from previous shares
-                        if share_list[i] == word:
-                            raise ShareAlreadyAddedError()
-            else:
-                # in case of advanced shamir recovery we only check 2 words
-                if i < 2:
-                    share_list = mnemonics[0].split(" ")
-                    if share_list[i] != word:
-                        raise IdentifierMismatchError()
+        # only if we know the backup type we can perform additional checks
+        if backup_type and not await check_mnemonic_validity(ctx, i, word, backup_type):
```

Comment:
fixed to 
backup_type is not None and ...

Reply:
ISTM this will fail with BIP39 now? or will it not get called?

Reply:
I believe you are correct, because `check_slip39_mnemonic_validity` requires some previous mnemonics to be present. I've made it more explicit in  d28e69c79db047318e2a2514ffb867bebe0d4682